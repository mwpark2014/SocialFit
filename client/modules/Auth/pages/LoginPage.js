import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormGroup, FormControl, Button,
  ControlLabel, Row, Col } from 'react-bootstrap';

// Import Style
// import styles from '../../components/Auth/PostListItem.css';

// Import Actions
import { loginUser } from '../AuthActions';

// Import Selectors
// import { getPost } from '../../AuthReducer';

export class LoginPage extends Component {
  constructor(props) {
    super(props);

    this.state = { name: '', username: '', email: '', password: '', password2: '' };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;
  
    this.setState({
      [name]: value,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    // Clear all the fields
    this.state = { name: '', username: '', email: '', password: '', password2: '' };
    this.props.loginUser({ username: this.state.username,
                   password: this.state.password });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <FormGroup>
          <Row>
            <Col md={2}>
              <ControlLabel>Username</ControlLabel>
            </Col>
            <Col md={4}>
            <FormControl type="text" name="username" placeholder="Username" // eslint-disable-line
              onChange={this.handleInputChange} value={this.state.username} required />
            </Col>
          </Row>
        </FormGroup>
        <FormGroup>
          <Row>
            <Col md={2}>
              <ControlLabel>Password</ControlLabel>
            </Col>
            <Col md={4}>
            <FormControl type="password" name="password" placeholder="Password" // eslint-disable-line
              onChange={this.handleInputChange} value={this.state.password} required />
            </Col>
          </Row>
        </FormGroup>
        <Button type="submit">Login</Button>
      </form>
    );
  }
}

function mapStateToProps(state) {
  return {
    errorMessage: state.auth.error,
    message: state.auth.message,
  };
}

LoginPage.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  loginUser: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { loginUser })(LoginPage);
