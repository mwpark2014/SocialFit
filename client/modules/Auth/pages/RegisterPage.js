import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import { Link } from 'react-router';

// Import Style
// import styles from '../../components/PostListItem/PostListItem.css';

// Import Actions
import * as actions from '../AuthActions';

// Import Selectors
// import { getPost } from '../../PostReducer';

export class RegisterPage extends Component {
  constructor(props) {
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.registerUser({ username: this.state.username,
                   name: this.state.name,
                   email: this.state.email,
                   password: this.state.password });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div>
          <label>Name</label>
          <input type="text" name="name" placeholder="Name" // eslint-disable-line
            onChange={this.handleInputChange} required />
        </div>
        <div>
          <label>Username</label>
          <input type="text" name="username" placeholder="Username" // eslint-disable-line
            onChange={this.handleInputChange} required />
        </div>
        <div>
          <label>Email</label>
          <input type="email" name="email" placeholder="Email" // eslint-disable-line
            onChange={this.handleInputChange} required />
        </div>
        <div>
          <label>Password</label>
          <input type="password" name="password" placeholder="Password" // eslint-disable-line
            onChange={this.handleInputChange} required />
        </div>
        <div>
          <label>Confirm Password</label>
          <input type="password" name="password2" placeholder="Password" // eslint-disable-line
            onChange={this.handleInputChange} required />
        </div>
        <button type="submit">Register</button>
        <br />
        Have you already registered? <Link to="/">Login</Link>
      </form>
    );
  }
}

function mapStateToProps(state) {
  return {
    errorMessage: state.auth.error,
    message: state.auth.message,
  };
}

RegisterPage.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};

export default connect(mapStateToProps, actions)(RegisterPage);
