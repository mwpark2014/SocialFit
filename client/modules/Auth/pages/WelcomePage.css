.welcomeContainer {
  margin: 0 auto;
  background-size: 500px auto;
  background: url('../../../assets/welcome-bg_1280x583.jpeg') no-repeat fixed center;
  height: 100%;
}

@media screen and (min-width: 1300px) {
  .welcomeContainer {
    background: url('../../../assets/welcome-bg_1920x1280.jpeg') no-repeat fixed center;
  }
}

.authContainer {
  margin: 50px 30px; 
  padding: 15px; 
  border: 1px solid #808080;
  border-radius: 5px;
  background-color: rgba(255, 255, 255, 0.8);
}