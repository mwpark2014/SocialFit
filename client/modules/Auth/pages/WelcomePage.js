import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Register from '../components/Register';
import Login from '../components/Login';
import { Row, Col } from 'reactstrap';

// Import Style
import styles from './welcomepage.css';

// Import Actions
import { registerUser, loginUser } from '../AuthActions';

// Import Selectors
import { getLoginButton } from '../../App/AppReducer';

export function WelcomePage(props) {
  const loginOrRegister = props.loginButton ?
    <Register
      errorMessage={props.errorMessage} message={props.message} registerUser={props.registerUser}
    /> :
    <Login
      errorMessage={props.errorMessage} message={props.message} loginUser={props.loginUser}
    />;
  return (
    <div className={styles.welcomeContainer}>
      <Row>
        <Col md={7} />
        <Col md={5}>
          <div className={styles.authContainer}>
            {loginOrRegister}
          </div>
        </Col>
      </Row>
    </div>
  );
}

function mapStateToProps(state) {
  return {
    errorMessage: state.auth.error,
    message: state.auth.message,
    authenticated: state.auth.authenticated,
    loginButton: getLoginButton(state),
  };
}

WelcomePage.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  authenticated: PropTypes.bool.isRequired,
  registerUser: PropTypes.func.isRequired,
  loginUser: PropTypes.func.isRequired,
  loginButton: PropTypes.bool.isRequired,
};

export default connect(mapStateToProps, { registerUser, loginUser })(WelcomePage);
