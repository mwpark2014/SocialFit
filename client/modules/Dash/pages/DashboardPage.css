.container {
  width: 100%;
  height: 100%;
  box-sizing: border-box;
}

.title {
  font-size: 18pt;
  font-weight: bold;
}