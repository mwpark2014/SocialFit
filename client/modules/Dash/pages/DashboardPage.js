import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

// Import Style
// import styles from '../../components/Auth/PostListItem.css';

// Import Actions
import { protectedTest } from '../../Auth/AuthActions';

// Import Selectors
// import { getPost } from '../../AuthReducer';

<<<<<<< HEAD
export function DashboardPage() {
  return (
    <div>
      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin ornare lorem mi, id efficitur ipsum faucibus vel. Vivamus quis vulputate lorem. Suspendisse suscipit suscipit tortor, venenatis ornare erat egestas ac. Nam blandit consequat turpis sed dictum. Aliquam ut diam vel ante pellentesque dictum ut ullamcorper justo. Proin et tempor velit, quis vehicula nulla. In viverra tellus ac est efficitur, sed pulvinar sem bibendum. Curabitur nibh tortor, dictum eu porta vitae, facilisis nec lacus.</p>
      <p>Vestibulum non felis pretium nulla scelerisque elementum. Praesent et mattis elit. Fusce congue dui lacus, sit amet pulvinar justo vehicula eu. Phasellus posuere tortor nunc, vel sagittis tortor interdum sed. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent vulputate nec odio ut fermentum. Donec accumsan augue nec nunc tincidunt auctor. Nam euismod ornare orci vel iaculis. Proin suscipit aliquet diam, id varius quam aliquet at. Nullam lacinia augue sed massa sollicitudin ornare.</p>
    </div>
  );
=======
export class DashboardPage extends Component {
  constructor(props) {
    super(props);

    this.props.protectedTest();
  }

  render() {
    return (
      <div>Your dashboard!</div>
    );
  }
>>>>>>> master
}

function mapStateToProps(state) {
  return { content: state.auth.content };
}

DashboardPage.propTypes = {
  protectedTest: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { protectedTest })(DashboardPage);
