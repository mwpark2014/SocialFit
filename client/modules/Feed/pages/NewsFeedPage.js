import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Row, Col } from 'reactstrap';
import FeedPostList from '../components/FeedPostList';

// Import Style
// import styles from './NewsFeedPage.css';

// Import Actions
import { fetchPosts } from '../../Post/PostActions';

// Import Selectors
import { getPosts } from '../../Post/PostReducer';

export function NewsFeedPage(props) {
  return (
    <div>
      <FeedPostList
        data={props.data} fetchPosts={props.fetchPosts}
      />
    </div>
  );
}

function mapStateToProps(state) {
  return {
    content: state.auth.content,
    username: state.auth.user,
    data: getPosts(state),
  };
}

NewsFeedPage.propTypes = {
  content: PropTypes.string.isRequired,
  fetchPosts: PropTypes.func.isRequired,
  data: PropTypes.array.isRequired,
  username: PropTypes.string.isRequired,
  params: PropTypes.object.isRequired,
};

export default connect(mapStateToProps, { fetchPosts })(NewsFeedPage);
