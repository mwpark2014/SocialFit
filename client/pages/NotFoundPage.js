import React, { Component } from 'react';

export function NotFoundPage() {
  return (
    <div>
      <h1>404 - Page Not Found</h1>
      <p>Sorry, the page you were looking for cannot be found!</p>
    </div>
  );
}

export default NotFoundPage;
