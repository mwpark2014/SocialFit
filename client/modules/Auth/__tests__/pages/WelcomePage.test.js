import React from 'react';
import { WelcomePage } from '../../pages/WelcomePage';
import renderer from 'react-test-renderer';

const registerUser = jest.fn(); //eslint-disable-line

const loginUser = jest.fn(); //eslint-disable-line

describe('Test render & styles', () => {
  test('WelcomePage snapshot', () => {
    const component = renderer.create(
      <WelcomePage
        errorMessage={''} message={''} registerUser={registerUser}
        loginUser={loginUser} authenticated loginButton
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(); // eslint-disable-line

    tree.props.authenticated = false;
    expect(tree).toMatchSnapshot(); // eslint-disable-line

    tree.props.loginButton = false;
    expect(tree).toMatchSnapshot(); // eslint-disable-line

    tree.props.authenticated = true;
    expect(tree).toMatchSnapshot(); // eslint-disable-line
  });
});
