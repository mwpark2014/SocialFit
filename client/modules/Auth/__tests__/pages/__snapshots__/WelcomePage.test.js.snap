// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render & styles WelcomePage snapshot 1`] = `
<div
  className="welcomeContainer"
>
  <div
    className="row"
  >
    <div
      className="col col-md-7"
    />
    <div
      className="col col-md-5"
    >
      <div
        className="authContainer"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Name
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Username
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="username"
                  onChange={[Function]}
                  placeholder="Username"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Email
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Password
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Confirm Password
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="password2"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col col-md-7 offset-md-4"
            >
              <button
                className="btn btn-secondary btn-block"
                onClick={[Function]}
                type="submit"
              >
                Register
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test render & styles WelcomePage snapshot 2`] = `
<div
  authenticated={false}
  className="welcomeContainer"
>
  <div
    className="row"
  >
    <div
      className="col col-md-7"
    />
    <div
      className="col col-md-5"
    >
      <div
        className="authContainer"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Name
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Username
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="username"
                  onChange={[Function]}
                  placeholder="Username"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Email
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Password
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Confirm Password
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="password2"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col col-md-7 offset-md-4"
            >
              <button
                className="btn btn-secondary btn-block"
                onClick={[Function]}
                type="submit"
              >
                Register
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test render & styles WelcomePage snapshot 3`] = `
<div
  authenticated={false}
  className="welcomeContainer"
  loginButton={false}
>
  <div
    className="row"
  >
    <div
      className="col col-md-7"
    />
    <div
      className="col col-md-5"
    >
      <div
        className="authContainer"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Name
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Username
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="username"
                  onChange={[Function]}
                  placeholder="Username"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Email
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Password
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Confirm Password
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="password2"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col col-md-7 offset-md-4"
            >
              <button
                className="btn btn-secondary btn-block"
                onClick={[Function]}
                type="submit"
              >
                Register
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test render & styles WelcomePage snapshot 4`] = `
<div
  authenticated={true}
  className="welcomeContainer"
  loginButton={false}
>
  <div
    className="row"
  >
    <div
      className="col col-md-7"
    />
    <div
      className="col col-md-5"
    >
      <div
        className="authContainer"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Name
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Username
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="username"
                  onChange={[Function]}
                  placeholder="Username"
                  required={true}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Email
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Password
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="form-group"
          >
            <div
              className="row"
            >
              <div
                className="col col-md-4"
              >
                <label
                  className=""
                  htmlFor={undefined}
                >
                  Confirm Password
                </label>
              </div>
              <div
                className="col col-md-7"
              >
                <input
                  className="form-control"
                  name="password2"
                  onChange={[Function]}
                  placeholder="Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col col-md-7 offset-md-4"
            >
              <button
                className="btn btn-secondary btn-block"
                onClick={[Function]}
                type="submit"
              >
                Register
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
